### DISCUSSION TITLE
A notification with a number of new articles instead of a notification per new article

### DISCUSSION THREAD

**Discussion by flameshikari** [jocmp/capyreader#554] [Feature Requests]
Hello!

First of all, I would like to thank you for this app. The app looks exactly as I dreamed of, a modern, uncluttered design with the ability to customize it a bit, and most importantly for me, it pulls up the content of articles better than many other readers. The last time I used Palabre for a long time, but then it stopped updating and started working poorly on modern versions of Android and looks off. I was looking for an app like it for a long time, and only recently came across yours. It's also amazing that it's open source and how quickly the app is developing. In general, success in development and great popularity to you and the app! Of course, I would like to participate in the development or support it somehow, but I have no experience in developing mobile apps, as well as no opportunity to donate for some reason. 🥲

Well, according to the title of the discussion, I'm suggesting a feature that will notify about a number of new articles in a single notification instead a notification per article grouped by feeds.

Current behavior:

![photo_2024-11-23_21-40-48](https://github.com/user-attachments/assets/12703c23-de33-4712-9525-768eae1e9b7a)

Suggested behavior:

![image](https://github.com/user-attachments/assets/892040a9-e2ec-472a-88e0-291e4e11addd)

**Comment by jocmp** [comment]
Hey @flameshikari. Thanks for the feature request and the kind words.

I really like the idea of a summary notification and grouping everything as much as possible. There are a few tweaks I can make. 

I want to keep the article dropdown so that users can click through to individual results much like an email or text message notification. But I think both the summary "New 10 article to read" and the dropdown might be possible.

**Reply by PhilC813** [reply]
> I want to keep the article dropdown so that users can click through to individual results much like an email or text message notification. But I think both the summary "New 10 article to read" and the dropdown might be possible.

This would indeed be the best implementation. It would also improve the current behavior, where a tap on a notification stack for a given feed doesn't do anything; a tap on the group should open the unread view of that given feed.

(Maybe some users will still complain about being bombarded with individual notifications instead of getting a summary every X hours, but considering that Capy is still young, these specific features are probably not critical to implement)

**Comment by jocmp** [comment]
I added some changes with [2024.12.1088-dev](https://github.com/jocmp/capyreader/releases/tag/2024.12.1088-dev). I wasn't able to include both a summary notification and group child articles, but the group will always show the number of new articles available at the top-end of the notification.

<img src="https://github.com/user-attachments/assets/91e86d27-f311-4b16-bc31-411fc02624ba" width="300px">

**Comment by OrkoGrayskull** [comment]
It's working great. Thank you!

A suggestion for improvement: I don't know if it's possible. But can the notifications disappear as soon as you open/read the corresponding article or mark all articles as read? Maybe Android doesn't allow this at all.

**Reply by jocmp** [reply]
It's something I want to do if it's possible. Android has a concept of a "delete intent" that I want to look into with https://github.com/jocmp/capyreader/issues/462 to add a "mark read" button to each notification. I'm hoping that will also work the other way like you said where marking an article as read will clear its notification.
